### PR TITLE
Fix defaulting to Pandas for empty Modin objects

### LINKED ISSUE #2833: to_numeric raises AttributeError with empty Series
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 20.04 LTS x86_64
- **Modin version** (`modin.__version__`): 0.9.0
- **Python version**: Python 3.8.5
- **Code we can use to reproduce**:

```python
import modin.pandas as pd
series = pd.Series([])
pd.to_numeric(series)   # raises AttributeError
```

Traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/general.py", line 344, in to_numeric
    return arg._to_numeric(errors=errors, downcast=downcast)
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/base.py", line 2830, in default_handler
    return self._default_to_pandas(item, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/base.py", line 377, in _default_to_pandas
    result = getattr(getattr(pandas, type(self).__name__), op)(
AttributeError: type object 'Series' has no attribute '_to_numeric'
```

### Describe the problem
When calling `to_numeric` on an empty `modin.pandas.series.Series` instance, an `AttributeError` is raised after warning that the default pandas implementation will be used.

This does not occur when the series is non-empty:

```
>>> pd.to_numeric(pd.Series([1, 2, 3]))
UserWarning: Distributing <class 'list'> object. This may take some time.
0    1
1    2
2    3
dtype: int64
>>> pd.to_numeric(pd.Series([]))
UserWarning: `Series._to_numeric` for empty DataFrame defaulting to pandas implementation.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/general.py", line 344, in to_numeric
    return arg._to_numeric(errors=errors, downcast=downcast)
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/base.py", line 2830, in default_handler
    return self._default_to_pandas(item, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/modin/pandas/base.py", line 377, in _default_to_pandas
    result = getattr(getattr(pandas, type(self).__name__), op)(
AttributeError: type object 'Series' has no attribute '_to_numeric'
```

This occurs using ray or dask.

### LINKED ISSUE #2839: Querying empty dataframes throws exceptions
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 20.04
- **Modin version** (`modin.__version__`): 0.9.0
- **Python version**: 3.8.5
- **Code we can use to reproduce**:

```
import modin.pandas as pd
df = pd.DataFrame(columns=['a', 'b'])
df[(df.a == 1) & (df.b == 2)]
```
Result:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...python3.8/site-packages/modin/pandas/series.py", line 147, in __and__
    new_self, new_other = self._prepare_inter_op(other)
  File "...python3.8/site-packages/modin/pandas/base.py", line 2830, in default_handler
    return self._default_to_pandas(item, *args, **kwargs)
  File "...python3.8/site-packages/modin/pandas/base.py", line 377, in _default_to_pandas
    result = getattr(getattr(pandas, type(self).__name__), op)(
AttributeError: type object 'Series' has no attribute '_prepare_inter_op'
```
### Describe the problem

This just returns an empty dataframe in vanilla pandas, but breaks down loudly in Modin.

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?
* Fix defaulting to Pandas for empty Modin objects - only default if Pandas class we mimick has the attribute being called
* Allow potential renaming (or inheritance) of mimicking classes by introducing `._pandas_class` class attribute
* Slightly improve `_get_axis_number()` (plus add proper docstring to it)

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2839
- [x] Resolves #2833
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/base.py
Comment (devin-petersohn) [inline modin/pandas/base.py:2813]: Why does this need to move/change?

**Review by devin-petersohn — COMMENTED** [review]
Overall, I like the changes, left a couple of comments.

File: modin/pandas/base.py
Comment (vnlitvinov) [reply modin/pandas/base.py:464]: I thought the parameter was obvious here just from description, thus didn't include it. I can add it if you wish, though.

File: modin/pandas/base.py
Comment (vnlitvinov) [reply modin/pandas/base.py:2813]: I doesn't _exactly_ need to change (though it could help with inheriting `BasePandasDataset` from something else, not `object`, which has `__getattribute__()` defined - a proper inheritance is you defer to your ancestor).

As for moving - this is a small optimization, as we're getting the attribute twice in an ordinary case - once to check if it's a method, and for a second time to return the result. There's no need to get it twice, it shouldn't change between the calls :)

File: modin/pandas/base.py
Comment (devin-petersohn) [reply modin/pandas/base.py:464]: Yes, for consistency sake I would prefer to add it.

File: modin/pandas/base.py
Comment (devin-petersohn) [reply modin/pandas/base.py:2813]: That makes sense.

**Review by devin-petersohn — COMMENTED** [review]
Thanks @vnlitvinov after the docs are update it will be good to merge.

File: modin/pandas/base.py
Comment (vnlitvinov) [reply modin/pandas/base.py]: Comment expanded.

**Review by devin-petersohn — APPROVED** [review]

**Pull request merged**
